src/Generator/PropertyGenerator.php

```diff
@@ -21,12 +21,40 @@ class PropertyGenerator extends AbstractMemberGenerator
 
     protected bool $isConst = false;
 
+    protected ?TypeGenerator $type = null;
+
     protected ?PropertyValueGenerator $defaultValue = null;
 
     private bool $omitDefaultValue = false;
 
+    /**
+     * @param  PropertyValueGenerator|string|array|null  $defaultValue
+     * @param  int|int[]                                 $flags
+     */
+    public function __construct(
+        ?string $name = null,
+        $defaultValue = null,
+        $flags = self::FLAG_PUBLIC,
+        ?TypeGenerator $type = null
+    ) {
+        parent::__construct();
+
+        if (null !== $name) {
+            $this->setName($name);
+        }
+        if (null !== $defaultValue) {
+            $this->setDefaultValue($defaultValue);
+        }
+        if ($flags !== self::FLAG_PUBLIC) {
+            $this->setFlags($flags);
+        }
+        if (null !== $type) {
+            $this->setType($type);
+        }
+    }
+
     /** @return static */
-    public static function fromReflection(PropertyReflection $reflectionProperty)
+    public static function fromReflection(PropertyReflection $reflectionProperty, bool $enablePropertyTypeHint = false)
```

Comment:
Boolean flag to be removed: if we're reflecting a symbol, and that symbol already has type declarations, then we're supposed to generate types for it (and also by default, like with all other `::fromReflection()` methods

Reply:
Who said we need to be backward compatible? 😂
If I remove this somebody gonna jump out and said it's not BC...

Reply:
Let's say some weirdo took a class in PHP 7.4 standard and trying to generate code for, say, PHP 7.1 (I was gonna say 5 😂), then this flag shows it's value.

Reply:
Not a problem - either you remove it, or I remove it before merging :P

Reply:
To remove this flag, I have to modify the unit test @samsonasik [objected before](https://github.com/laminas/laminas-code/pull/123#pullrequestreview-829578913)

If you say so sir, I'll get it removed straight away.

Reply:
@samsonasik no offense, but let's get rid of this stuff :D

Reply:
Just removed the flag as per requested by @Ocramius